benchmark/reformat_input.py

```diff
@@ -337,19 +339,192 @@ def reformat_treepuzzle(gene_tree,
     msa_fa.write(output_msa_phy_fp, format='phylip')
 
 
+def _merge_genbank_seqs(genbank_fp):
+    """ Merge one to multiple sequences in a GenBank file into one.
+
+    Parameters
+    ----------
+    genbank_fp: string
+        file path to genome in GenBank format
+
+    Returns
+    -------
+    tuple of (
+        skbio.Sequence,
+            Genome sequence and metadata
+        dict of { list of [ string, int, int, string ] }
+            Gene name : translation, start, end, and strand
+    )
+    """
+    loci = []
+    nucl_seq = ''
+    genes = {}
+    nseq = 0  # number of nucleotide sequences
+    with open(genbank_fp, 'r') as input_f:
+        for line in input_f:
+            if line.startswith('//'):
+                nseq += 1
+    abs_pos = 0  # absolute position in concantenated nucleotide sequence
+    for i in range(nseq):
+        gb = Sequence.read(genbank_fp, seq_num=i+1, format='genbank')
+        locus_name = gb.metadata['LOCUS']['locus_name']
+        size = gb.metadata['LOCUS']['size']
+        loci.append([locus_name, size])
+        nucl_seq += str(gb)
+        for feature in gb.interval_metadata.features:
+            if feature['type_'] == 'CDS' and 'protein_id' in feature:
+                protein_id = feature['protein_id'].replace('\"', '')
+                if protein_id not in genes:
+                    translation = feature['translation'].replace(' ', '') \
+                        .replace('\"', '')
+                    strand = '-' if feature['rc_'] else '+'
+                    loc = gb.interval_metadata.features[feature]
+                    start = loc[0][0] + abs_pos + 1
+                    end = loc[0][1] + abs_pos
+                    genes[protein_id] = [translation, start, end, strand]
+        abs_pos += int(size)
+    gb = DNA(nucl_seq)
+    gb.metadata['LOCUS'] = {'locus_name': 'locus001', 'size': len(nucl_seq),
+                            'unit': 'bp', 'shape': 'circular',
+                            'division': 'CON', 'mol_type': 'DNA',
+                            'date': '01-JAN-1900'}
+    gb.metadata['id'] = 'locus001'
+    return (gb, genes)
+
+
+def _write_genebank_file(gb,
+                         genes,
+                         output_dir):
+    """ Write output genome to a GenBank file.
+
+    Parameters
+    ----------
+    gb: skbio.Sequence,
+        Genome sequence and metadata
+    genes: dict of { list of [ string, int, int, string ] }
+        Gene name : translation, start, end, and strand
+    output_dir: string
+        output directory path
+
+    Notes
+    -----
+    The reason this script does not use skbio's GenBank write function, is
+    that EGID cannot parse skbio-written GenBank format. Perhaps the reason
+    is the differential pattern of leading white spaces in the feature table.
+    """
+    output_f = open(join(output_dir, 'id.gbk'), 'w')
+    tmp_fp = join(output_dir, 'id.tmp')
+    DNA.write(gb, tmp_fp, format='genbank')
```

Comment:
Thanks for noting this. Yes it sounds odd. The logic is: scikit-bio can generate a GenBank file, but its format is not exactly compatible with EGID. Specifically, this incompatibility is in the feature table. Therefore, my solution was:
1) Use scikit-bio to write a GenBank structure without feature table to a temporary file.
2) Open this temp file, read the first line (locus table), and write into the output GenBank file.
3) Use Python to write the feature table into the output GenBank file.
4) Open the temp file again, read the rest lines (genome sequence), and write into the output GenBank file.
Does this sound a reasonable plan?